### PR TITLE
Revise IIRuncFilter to enable streaming

### LINKED ISSUE #154: Extend IIRuncFilter() to colored noise inputs
The function *`IIRuncFilter`* in _`uncertainty.propagate_filter`_ does currently only work for white noise input signals. The corresponding paper also applies for colored noise. In addition, that the output can be covariance matrix needs to be addressed. Currently it's only point-wise standard deviations.

This is dealt with in #95

### PR DESCRIPTION
**Pull request by BjoernLudwigPTB**
We introduced the ability to define an initial state for the IIRuncFilter to enable efficient subsequent calls.

At the same time we resolve #154 .

### CODE REVIEW

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- examples/Digital filtering/speed_comparison_filter.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/PTB-PSt1/PyDynamic/pullRequest?prid=5056465&bid=16644433)

**Comment by BjoernLudwigPTB** [comment]
@mgrub There you go! I did not delete your branch IIRuncFilter_realtime, which you probably should do now. By the way, it was necessary to cherry-pick one more commit from add_wavelet, which introduced [import time as time_measure](https://github.com/PTB-PSt1/PyDynamic/blob/7049188ab5d10e28f491d99722de398c50be0846/test/test_propagate_filter.py#L7) into your tests.

**Comment by mgrub** [comment]
@BjoernLudwigPTB : Thanks a lot! I have now deleted the IIRuncFilter_realtime branch locally as well on github.

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- examples/Digital filtering/speed_comparison_filter.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/PTB-PSt1/PyDynamic/pullRequest?prid=5056465&bid=16644433)

**Comment by mgrub** [comment]
I will close this pull request and open a new one, that requests a merge into the master / prepare_major_release_2.0 .

File: PyDynamic/uncertainty/propagate_filter.py [hunk 1]
```diff
@@ -171,95 +181,298 @@ def FIRuncFilter(y, sigma_noise, theta, Utheta=None, shift=0, blow=None, kind="c
...
+
+def _adjust_filter_coefficients(b, a, Uab):
+    """
+    Bring b, a to the same length and adjust Uab accordingly
+    """
+    # length difference of both filters
+    d = len(a) - len(b)
+
+    # adjust filter coefficient uncertainties
+    if isinstance(Uab, np.ndarray):
```
Comment (BjoernLudwigPTB) [inline PyDynamic/uncertainty/propagate_filter.py:456]: What else do you expect? We should mention this somewhere here or motivate this check.

File: test/test_propagate_filter.py [hunk 2]
```diff
@@ -2,55 +2,57 @@
...
-    x = rect(time,100*Ts,250*Ts,1.0,noise=sigma_noise)
+    x = rect(time, 100 * Ts, 250 * Ts, 1.0, noise=sigma_noise)
 
     # apply uncertain FIR filter (GUM formula)
     for blow in [None, b2]:
         y, Uy = FIRuncFilter(x, sigma_noise, b1, Ub, blow=blow, kind="float")
         assert len(y) == len(x)
         assert len(Uy) == len(x)
 
+
```
Comment (BjoernLudwigPTB) [inline test/test_propagate_filter.py:55]: Shouldn't we test those private functions separately?

**Comment by BjoernLudwigPTB** [comment]
Looks like this [fancy notation for formatted string was just introduced in Python 3.6](https://docs.python.org/3.6/whatsnew/3.6.html#whatsnew36-pep498). Thus we should reside to the `.format` notation.

**Pull request merged**